data-prepper-plugins/common/src/main/java/org/opensearch/dataprepper/common/utils/CaffeineCache.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.common.utils;
+import com.google.common.annotations.VisibleForTesting;
+
+import org.opensearch.dataprepper.model.event.Event;
+
+import com.github.benmanes.caffeine.cache.RemovalCause;
+import com.github.benmanes.caffeine.cache.RemovalListener;
+import com.github.benmanes.caffeine.cache.Cache;
+import com.github.benmanes.caffeine.cache.stats.CacheStats;
+import com.github.benmanes.caffeine.cache.Caffeine;
+import com.github.benmanes.caffeine.cache.Weigher;
+
+import java.lang.instrument.Instrumentation;
+import java.util.concurrent.TimeUnit;
+import java.util.Arrays;
+
+public class CaffeineCache<K, V> {
+
+    public static final int OVERHEAD_PER_CACHE_ENTRY = 32;
+
+    class ByteArrayValueWeigher implements Weigher<K, V> {
+
+        @Override
+        public int weigh(K key, V value) {
+            return value.toString().length() + key.toString().length();
+        }
+    }
+
+    private final Cache<K, V> cache;
+    private final long maxBytes;
+    private final int maxEntries;
+    private int curEntries;
+    private long curBytes;
+    private boolean strictLRU;
+
+    public CaffeineCache(int maxEntries, long maxBytes, int ttlSeconds, boolean strictLRU) {
```

Comment:
We should probably avoid exposing the constructor and use a builder pattern for this, similar to Caffeine's builder pattern.